pallets/manta-pay/src/lib.rs

```diff
@@ -202,8 +202,18 @@ pub mod pallet {
         pub fn to_private(origin: OriginFor<T>, post: TransferPost) -> DispatchResultWithPostInfo {
             let origin = ensure_signed(origin)?;
             for source in post.sources.iter() {
-                ensure!(asset_value_decode(*source) > 0u128, Error::<T>::ZeroTransfer);
+                ensure!(
+                    asset_value_decode(*source) > 0u128,
+                    Error::<T>::ZeroTransfer
+                );
             }
+            ensure!(
+                post.sources.len() == 1
```

Comment:
i think 2 in and 2 out is also valid, though

Reply:
in the previous version, when private transfer need one zero coin to fillout 2 in, it'll also invoke to_private extrinc, but current version already get rid this part. so to_private is only for public token to private token. and it only has 1 sources, 1 receiver posts. For private transfer, it have 2 sender posts and 2 receiver posts.

Reply:
Ah yeah, saw it here as well https://github.com/Manta-Network/manta-rs/blob/6572219b4793a5fed9ae4c8ecc1281eadbc038a4/manta-accounting/src/transfer/canonical.rs#L95